lib/auth/auth.go

```diff
@@ -3345,3 +3349,60 @@ func isHTTPS(u string) error {
 
 	return nil
 }
+
+// AddClusterCAPoolsToTLSConfig return a copy of tls.Config and set the  ClientCAs
+// to the cluster CA Pool.
+func AddClusterCAPoolsToTLSConfig(tlsConfig *tls.Config, ap AccessPoint, currentClusterName string, log logrus.FieldLogger) func(*tls.ClientHelloInfo) (*tls.Config, error) {
```

Comment:
Nit: This does not really add anything by itself but returns a function that does it rather. Maybe a name like `WithClusterCAs` or something like that would work a little better?